scipy/optimize/tests/test__differential_evolution.py

```diff
@@ -519,10 +519,39 @@ def test_random_generator(self):
             differential_evolution(self.quadratic,
                                    [(-100, 100)],
                                    polish=False,
-                                   seed=rng,
+                                   rng=rng,
                                    tol=0.5,
                                    init=init)
 
+    def test_rng_seed_spec007(self):
+        # spec007 involves the transition of RandomState-->Generator
+        # rng is the new name
+        differential_evolution(self.quadratic,
+                               [(-100, 100)],
+                               polish=False,
+                               rng=1,
+                               tol=0.5)
+        # should still be allowed to pass `seed`
+        differential_evolution(self.quadratic,
+                               [(-100, 100)],
+                               polish=False,
+                               seed=1,
+                               tol=0.5)
+        with assert_raises(TypeError):
+            # can't pass both seed and rng
+            differential_evolution(self.quadratic,
+                                   [(-100, 100)],
+                                   polish=False,
+                                   seed=1,
+                                   rng=1,
+                                   tol=0.5)
+            # use of rng=RandomState should give rise to an error.
+            differential_evolution(self.quadratic,
```

Comment:
I think these need to be in separate context managers as the test will still pass if only one these raises an error

Reply:
Yup. Let's fix that in the next PR like this.